netket/optimizer/qgt/qgt_jacobian_pytree.py

```diff
@@ -70,8 +72,8 @@ def QGTJacobianPyTree(
     from netket.vqs.exact import ExactState
 
     if isinstance(vstate, ExactState):
-        samples = vstate._all_states
-        pdf = vstate.probability_distribution()
+        samples = split_array_mpi(vstate._all_states)
+        pdf = split_array_mpi(vstate.probability_distribution())
```

Comment:
@nikita-astronaut 
The `split_array_mpi ` code pads the array by appending the first few elements to the end if the size of the array ( `hi.n_states` in this case) is not divisible by the number of ranks. 
Here it would however be necessary to either pad the pdf with zeros such that the expectation values are correct, or not pad at all and just have a different shape on the last rank.

Reply:
You are right, I overlooked that. I only tested the case where it is divisible

Reply:
fixed that